packages/adapter-mongoose/lib/adapter-mongoose.js

```diff
@@ -88,6 +88,11 @@ class MongooseAdapter extends BaseKeystoneAdapter {
 
     schema.add({ [leftFkPath]: {} });
     schema.add({ [rightFkPath]: {} });
+
+    // create 2 way indexes for reference fields
+    schema.index({ [leftFkPath]: 1, [rightFkPath]: 1 });
```

Comment:
is this creating combined index? 

why not create index for each field (left and right) separately without combining them, they are non unique so it would not have issues.

Reply:
@gautamsi yep, I was also thinking if we really need to make it a compound index. But I found it useful, see this case:
Users and Projects have N:N relation.
```
query() {
    allProjects() {
       users() {
       }
    }
}
```
Currently keystone will request by project_left_id only, but if we add additional filter to users:
```
query() {
    allProjects(where: { id: 1 }) {
       users(where: { id: 2 }) {
       }
    }
}
```
mongodb will use IXSCAN only by one field Project_left_id, but to retrieve nested users with `users(id = 1)` it has to apply additional in memory filter because the second field is not indexed. So making compound index would resolve that. Anyway the second field is not going to take a lot of space.
Let me know if I am at the wrong direction.

Reply:
I meant there could be two individual indexes. one for each field. So they both be indexed.

Reply:
@gautamsi I am creating for both of them:
```
schema.index({ [leftFkPath]: 1, [rightFkPath]: 1 });
schema.index({ [rightFkPath]: 1, [leftFkPath]: 1 });
```

Reply:
why not do this

```js
schema.index({ [leftFkPath]: 1 });
schema.index({ [rightFkPath]: 1 });
```

Reply:
@gautamsi fixed

Reply:
thanks, hopefully @timleslie likes the split index